### PR TITLE
The connector skeleton

### PR DESCRIPTION
**Pull request by SiliconMeeple**
This starts the connector application up, connects to a database and
healthchecks the connection periodically.

### CODE REVIEW

File: src/test/java/uk/gov/pay/connector/util/PostgresDockerRule.java [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+import java.util.Optional;
+
+public class PostgresDockerRule implements TestRule {
+
+    private static String host;
+    private static PostgresContainer container;
+
+    static {
+        try {
+            host = Optional.ofNullable(new URI(System.getenv("DOCKER_HOST")).getHost()).orElse("localhost");
```
Comment (acceleanu) [inline src/test/java/uk/gov/pay/connector/util/PostgresDockerRule.java]: Fails with NPE when DOCKER_HOST is not set.
Maybe check here explicitly for unix socket.

File: src/main/java/uk/gov/pay/connector/app/ConnectorApp.java [hunk 2]
```diff
@@ -0,0 +1,42 @@
+package uk.gov.pay.connector.app;
+
+import io.dropwizard.Application;
+import io.dropwizard.Configuration;
+import io.dropwizard.configuration.EnvironmentVariableSubstitutor;
+import io.dropwizard.configuration.SubstitutingSourceProvider;
```
Comment (acceleanu) [inline src/main/java/uk/gov/pay/connector/app/ConnectorApp.java:5]: clean unused imports

File: src/test/java/uk/gov/pay/connector/util/PostgresContainer.java [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+
+    public PostgresContainer(DockerClient docker, String host) throws DockerException, InterruptedException, IOException, ClassNotFoundException {
+        Class.forName("org.postgresql.Driver");
+
+        this.docker = docker;
+        this.host = host;
+        docker.pull(POSTGRES);
+
+        final HostConfig hostConfig = HostConfig.builder().publishAllPorts(true).build();
+        ContainerConfig containerConfig = ContainerConfig.builder().image(POSTGRES).hostConfig(hostConfig).env("POSTGRES_USER=" + DB_USERNAME, "POSTGRES_PASSWORD=" + DB_PASSWORD).build();
```
Comment (acceleanu) [inline src/test/java/uk/gov/pay/connector/util/PostgresContainer.java]: this would read better on separate lines for each method invocation

Comment (SiliconMeeple) [inline src/test/java/uk/gov/pay/connector/util/PostgresContainer.java]: If we're sticking to that as a convention, could we figure out a way to get IntelliJ to automatically format code that way?

Comment (acceleanu) [inline src/test/java/uk/gov/pay/connector/util/PostgresContainer.java]: thought we have agreed that we would use the IJ defaults

**Comment by acceleanu** [comment]
test

**Pull request merged**